### DISCUSSION TITLE
Is there a catch all for all routes inside a "main" route? /route/*

### DISCUSSION THREAD

**Discussion by nicholasfc** [vercel/next.js#16125] [Help] [answered]
I need to check if a non admin user is inside any /admin/* routes and redirect the user to a 404 page. Right now I can only do checking page by page

`if (router.pathname === '/admin/page1' && !userIsAdmin) {
      router.push('/404');
    }`

**Comment by hk86** [comment, marked as answer]
`if (router.pathname.startsWith('/admin/') && !userIsAdmin) { router.push('/404'); }`

Will this work for you?

**Reply by nicholasfc** [reply]
Yes, thanks a lot!
